scene/resources/visual_shader_nodes.cpp

```diff
@@ -1725,6 +1725,119 @@ VisualShaderNodeLinearSceneDepth::VisualShaderNodeLinearSceneDepth() {
 	simple_decl = false;
 }
 
+////////////// World Position from Depth
+
+String VisualShaderNodeWorldPositionFromDepth::get_caption() const {
+	return "WorldPositionFromDepth";
+}
+
+int VisualShaderNodeWorldPositionFromDepth::get_input_port_count() const {
+	return 0;
+}
+
+VisualShaderNodeWorldPositionFromDepth::PortType VisualShaderNodeWorldPositionFromDepth::get_input_port_type(int p_port) const {
+	return PORT_TYPE_SCALAR;
+}
+
+String VisualShaderNodeWorldPositionFromDepth::get_input_port_name(int p_port) const {
+	return "";
+}
+
+int VisualShaderNodeWorldPositionFromDepth::get_output_port_count() const {
+	return 1;
+}
+
+VisualShaderNodeWorldPositionFromDepth::PortType VisualShaderNodeWorldPositionFromDepth::get_output_port_type(int p_port) const {
+	return PORT_TYPE_VECTOR_3D;
+}
+
+String VisualShaderNodeWorldPositionFromDepth::get_output_port_name(int p_port) const {
+	return "world position";
+}
+
+bool VisualShaderNodeWorldPositionFromDepth::has_output_port_preview(int p_port) const {
+	return false;
+}
+
+String VisualShaderNodeWorldPositionFromDepth::generate_global(Shader::Mode p_mode, VisualShader::Type p_type, int p_id) const {
+	return "uniform sampler2D " + make_unique_id(p_type, p_id, "depth_tex") + " : hint_depth_texture;\n";
```

Comment:
I'm worried about every `WorldPositionFromDepth` node adding another sampler2D declaration to the generated shader.

Is there any performance cost to having lots of duplicate uniform sampler2D, for example maybe I have multiple if statements or I want to sample at different coordinates?

Also, it's strongly recommended that `repeat_disable, filter_nearest` be used together with `hint_depth_texture`. It's unfortunate that this isn't the default, I believe... but it's what you are supposed to do to avoid artifacts at the screen edges.

Another option would be to make a separate node for the depth texture, and then have users connect the nodes. That way, we can avoid unnecessary sampler2D declarations.

Reply:
It's impossible to do this without having UX issues or compatability breaking changes. We already have the same code in the LinearSceneDepth node. Also imho if you really want performant code you write it and clean it up and only use one node that has lots of connections. This should be perhaps better solved by an extra node you can tunnel other nodes through like in MaterialMaker.
I did add the other flags though like you suggested.